### DISCUSSION TITLE
WiFi over WiFi issue (not related to technicolor)

### DISCUSSION THREAD

**Discussion by fa1rid** [hack-technicolor/hack-technicolor#119] [General]
Hi Guys,

Sorry for posting an unrelated question here but I'm really annoyed from this issue and I need your opinion about it.

So, I have tested several times the following scenario (with different hardware) and I'm sure there's some kind of issue which I don't understand.

1st WiFi AP > WiFi Client > NAT > 2nd WiFi AP > Final WiFi Client (throughput is less than 1Mbps)

If both first and second AP uses 2.4GHz, then the speed becomes very low ~1Mbps, however if you replace second AP with 5GHz and keep the first one 2.4GHz, then there's no problem at all. What's going on? Anyone has any clue why this is happening please?

Thanks

**Comment by kevdagoat** [comment]
STA mode on routers is not very fast, make sure that the WiFi client has max channel width and is on the same channel as the main AP

**Comment by fa1rid** [comment]
Thanks for replying, I'm not sure if everything you said makes sense.
The problem happens only when there's NAT and then 2.4GHz AP in this specific setup.

In the following cases it's normal and no speed issues:
1- if you make 2nd AP run on 5GHz.
2- if you connect directly through Ethernet to the NAT mentioned.
3- If you remove the NAT and bridge 2nd WiFi AP wit the previous WiFi Client.

**Comment by FrancYescO** [comment]
you should be clear about the hardware used and check the Connection state of the "WiFi Client", probably it will fallback on only G mode or something like this, take clear that using the same radio for "WiFi Client" and "2nd WiFi AP" will make the speed at half of the radio capability

**Comment by fa1rid** [comment]
regarding hardware I tried everything, a router, a PC, a RPi..
Several wifi clients and wifi adapter has been under this test, all separate radios.
Please try yourself the same setup and let me know.
Thanks

**Comment by FrancYescO** [comment]
You tried everything at what point of the chain?

I have no issue reaching 20+Mbps on the farest client over an 2.4Ghz repeater bridge (N mode)

https://openwrt.org/docs/guide-user/network/wifi/relay_configuration

**Comment by fa1rid** [comment]
Every device in the chain! I have a couple of routers, ranger extenders, PCs, RPi, phones, WiFi USB adapters.
Don't use a bridge, use NAT. I mentioned a bridge works fine. I tired both N and G modes, same thing no difference.

**Comment by FrancYescO** [comment]
yes but as asked before at which speed and mode the "WiFi Client" is connected?

**Comment by fa1rid** [comment]
They are all connected  on WiFi 4 N mode (2.4GHz)

**Comment by FrancYescO** [comment]
at wich speed/dbi/noise level?

**Comment by fa1rid** [comment]
The first WiFi client can receive internet speed of 20Mbps at around -75dbi. Then it goes to NAT > AP > client receiving 1Mbps at around -50dbi.
I didn't test noise level.

**Comment by fa1rid** [comment]
Sorry the first one can go minimum 30Mbps not 20.

I tried another main AP (first one in chain), and I am able to get around 8Mbps, still far from 30Mbps which I'm getting on 5GHz.

**Comment by fa1rid** [comment]
The strange thing is that first APs are able to provide 30Mbps to the other end using Ethernet from NAT or (5GHz 2nd AP).

**Comment by FrancYescO** [comment]
looks at you location 2.4Ghz bands are just overpopulated and disturbed, you should try to get more technical infos on the connection signal.

**Comment by fa1rid** [comment]
Sorry what?
there's only around 3 other wifi networks in the air, all using different channels. Also when connecting to any device directly there's no problem at all knowing that everything else is still transmitting in the air as it is, so I'm sure there's no interference or something like that. I tried also 3 different locations, same thing.

**Comment by FrancYescO** [comment]
not only wifi uses the 2.4 bands and can cause interferences (so looking at what channels the nearest network are transmitting is not enough to know you are getting no interferences, and also, just 1network not respecting the standards can destroy the throughput of the yours)

if you have the possibility to not use 2.4ghz for a bridge just avoid it, is the worst choice.

**Comment by fa1rid** [comment]
> if you have the possibility to not use 2.4ghz for a bridge just avoid it, is the worst choice.

what do you mean? I'm not using a bridge.
Can you please try the same exact setup and tell me what speed you get?
If there's interference then I could notice it as I'm using many devices. Only this specific chain is an issue.
And as I said local connection to any part of the chain is fine except when the traffic has to pass throughout the entire chain.

I will do one more test locally without using the internet to be more sure it's not an ISP issue.

If you strongly believe it's interference issues, how can I be more sure?
Thanks

**Comment by FrancYescO** [comment]
"1st WiFi AP > WiFi Client > NAT > 2nd WiFi AP" this is basically a bridge to the final client (yeah using NAT not a real bridge).

There is nothing that i have to test, this is a pretty standard wifi repeater configuration (sometime with no NAT)
 and as i've said i have more sustained speed that 1Mbps on the "Final WiFi Client", all dependes on radios of devices used and signal quality.

you could try to do some sort of speedtest on "WiFi Client" or as i'm still saying, trying to read more technical details on the connection state ("WiFi Nurse" of technicolor devices is pretty detailed)

**Comment by fa1rid** [comment]
I tested locally from end to end with iperf3 and same thing it's slow. "WiFi Client" speed is normal and no issue as I already say even with NAT using Ethernet to or 5GHz there's no problem.

Can this be a data fragmentation issue?

**Comment by hiatylellah** [comment]
Hi farid, I need to communicate with you, I am also living at  @fa1rid 
I need your support for Technicolor modem similar to yours
